src/app/services/ws.service.spec.ts

```diff
@@ -85,6 +92,22 @@ describe('WebSocketService', () => {
     });
   });
 
+  describe('callAndSubscribe', () => {
+    it('should call and subscribe to updates', async () => {
+      const observablePools$ = of(pools);
+      jest.spyOn(service, 'callAndSubscribe').mockImplementationOnce((method: string) => {
```

Comment:
This doesn't really test `callAndSubscribe`, because you've overridden it are just checking that your mock was called correctly.